lib/xboxrt/libc_extensions/direct.c

```diff
@@ -0,0 +1,108 @@
+// SPDX-License-Identifier: MIT
+
+// SPDX-FileCopyrightText: 2023 ExoSkye
+
+// Part of Microsoft CRT
+
+#include <errno.h>
+#include <string.h>
+#include <stdbool.h>
+#include <windows.h>
+
+// Made referencing https://www.digitalmars.com/rtl/direct.html, retrieved 2023-06-10, copyrighted 1999-2018 by Digital Mars
+
+#define MAXPATH 80
+#define MAXDRIVE 3
+
+/*
+ * These silently succeed, since they wouldn't have any effect on the program at all
+ */
+
+int _chdir(char* path) {
+    return 0;
+}
+
+int _chdrive(int drive) {
+    return 0;
+}
+
+/*
+ * The Xbox has no concept of current working directory, so these can't work
+ */
+
+char* _getcwd(char* buffer, size_t length) {
+    errno = -EINVAL;
+    return NULL;
+}
+
+char* _getwd(char* path_name) {
+    errno = -EINVAL;
+    return NULL;
+}
+
+int _getdrive(void) {
+    errno = -EINVAL;
+    return 0;
+}
+
+/*
+ * There's no path variable on the Xbox, so this can't work
```

Comment:
The linked docs (although not really a good source) say the cwd is searched first.

Is this function ever used though?
For functions which aren't really used, I think it's better to leave them out, rather than broken stubs.

We should implement (or stub, if necessary) those functions which are actually used in software that people want to use / port.

Reply:
Oh shit, I didn't realise that, I guess I need to do more research into it

I've never see any usage of this in the wild, however

Reply:
This function doesn't exist in the `wine` project source code, so I'm going to remove it

Reply:
Resolved in 72ae765